components/raftstore/src/store/fsm/peer.rs

```diff
@@ -5974,15 +5998,41 @@ where
         }
     }
 
-    fn on_approximate_region_size(&mut self, size: u64) {
-        self.fsm.peer.approximate_size = Some(size);
+    fn on_approximate_region_size(&mut self, size: Option<u64>, may_split: Option<bool>) {
+        // if size is none, it means no estimated size
+        if size.is_some() {
+            self.fsm.peer.approximate_size = size;
+        }
+
+        if may_split.is_some() {
+            self.fsm.peer.may_split_size = may_split;
+        }
+
+        // if the region is truely splittable,
+        // may_skip_split_check should be false
+        if matches!(may_split, Some(true)) {
+            self.fsm.peer.may_skip_split_check = false;
+        }
         self.register_split_region_check_tick();
         self.register_pd_heartbeat_tick();
         fail_point!("on_approximate_region_size");
     }
 
-    fn on_approximate_region_keys(&mut self, keys: u64) {
-        self.fsm.peer.approximate_keys = Some(keys);
+    fn on_approximate_region_keys(&mut self, keys: Option<u64>, may_split: Option<bool>) {
+        // if keys is none, it means no estimated keys
+        if keys.is_some() {
+            self.fsm.peer.approximate_keys = keys;
+        }
+
+        if may_split.is_some() {
```

Comment:
write in `if let` clause with below?

Reply:
But with "if let",  it would then use "Some(may_split)" instead of may_split directly.